contracts/token/ERC1155/MultiToken.sol

```diff
@@ -1,35 +1,48 @@
+// SPDX-FileCopyrightText: 2022 ISKRA Pte. Ltd.
 // SPDX-License-Identifier: MIT
+// @author iskra.world dev team
+
 pragma solidity ^0.8.2;
 
-import "@openzeppelin/contracts/token/ERC1155/extensions/ERC1155URIStorage.sol";
-import "@openzeppelin/contracts/access/AccessControl.sol";
+import "@openzeppelin/contracts/access/Ownable2Step.sol";
 import "@openzeppelin/contracts/security/Pausable.sol";
+import "@openzeppelin/contracts/token/ERC1155/extensions/ERC1155URIStorage.sol";
 import "@openzeppelin/contracts/token/ERC1155/extensions/ERC1155Burnable.sol";
 import "@openzeppelin/contracts/token/ERC1155/extensions/ERC1155Supply.sol";
 
 // MultiToken features
 //  - ERC1155 basic functions
-//  - role base
-//  - mintable, not burnable
-//  - pausable
+//  - configurable for pausable, burnable (defined at construction)
 //  - defense for transferring tokens to the token contract itself
 contract MultiToken is
     ERC1155URIStorage,
-    AccessControl,
+    Ownable2Step,
     Pausable,
     ERC1155Burnable,
     ERC1155Supply
 {
-    bytes32 public constant URI_SETTER_ROLE = keccak256("URI_SETTER_ROLE");
-    bytes32 public constant PAUSER_ROLE = keccak256("PAUSER_ROLE");
-    bytes32 public constant MINTER_ROLE = keccak256("MINTER_ROLE");
+    bool public immutable pausable;
+    bool public immutable burnable;
 
-    mapping(uint256 => string) internal uris;
+    event ApprovalBurnPermission(
```

Comment:
nit:
How about changing the event name? 
- BurnPermissionApproval
- Approve(Authorize)BurnPermission
- GrantBurnPermission

Reply:
I think `BurnPermissionApproval` would be good.